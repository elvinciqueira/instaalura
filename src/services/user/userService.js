import {isStagingEnv} from '../../infra/env/isStagingEnv'
import HttpClient from '../../infra/HttpClient'
import {authService} from '../auth/authService'

const BASE_URL = isStagingEnv
  ? 'https://instalura-api-git-master.omariosouto.vercel.app'
  : 'https://instalura-api.omariosouto.vercel.app'

export const userService = {
  async getProfilePage(ctx, HttpClientModule = HttpClient, authServiceModule = authService) {
    const url = `${BASE_URL}/api/users/posts`
    try {
      const token = await authServiceModule(ctx).getToken()
      const response = await HttpClientModule(url, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })

      return {
        user: {
          totalLikes: 100,
        },
        posts: response.data,
      }
    } catch (err) {
      throw new Error('NÃ£o conseguimos pegar os posts')
    }
  },
}
