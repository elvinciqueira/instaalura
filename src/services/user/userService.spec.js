import {userService} from './userService'

const posts = [{
  _id: 'fake-id',
  photoUrl: 'image-url',
  filter: 'instagram-filter',
  description: 'awesome description'
}]


const authServiceModule = jest.fn(() => ({
  getToken: jest.fn(() => 'fake-token')
}))

const HttpClientModule = jest.fn()
  .mockImplementationOnce(() => Promise.resolve({ data: posts}))

describe('userService', () => {
  describe('getProfilePage()', () => {
    describe('when logged in', () => {
      it('returns user profile data', async () => {
        const expectedResponseData = {
          user: {
            totalLikes: 100
          },
          posts: [{
            _id: 'fake-id',
            photoUrl: 'image-url',
            filter: 'instagram-filter',
            description: 'awesome description'
          }]
        }
        const expectedToken = 'fake-token'
        
        const profilePage = await userService.getProfilePage(null, HttpClientModule, authServiceModule)

        expect(HttpClientModule).toHaveBeenCalledTimes(1)
        expect(authServiceModule).toHaveBeenCalledTimes(1)
        expect(authServiceModule().getToken()).toBe(expectedToken)
        expect(profilePage).toEqual(expectedResponseData)
      })
    })

    describe('when token is invalid', () => {
      it('throws an error', async () => {
        expect.assertions(1)
        
        try {
          await userService.getProfilePage(null, HttpClientModule)
        } catch (error) {
          expect(error).toMatchSnapshot()
        }
      })
    })
  })
})