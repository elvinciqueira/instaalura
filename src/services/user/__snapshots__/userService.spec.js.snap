// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userService getProfilePage() when token is invalid throws an error 1`] = `[Error: NÃ£o conseguimos pegar os posts]`;
